include/engine/data_watchdog.hpp

```diff
@@ -26,10 +27,23 @@ class DataWatchdog
 {
   public:
     DataWatchdog()
-        : shared_barriers{std::make_shared<storage::SharedBarriers>()},
-          shared_regions(storage::makeSharedMemory(storage::CURRENT_REGION)),
-          current_timestamp{storage::REGION_NONE, 0}
+        : active(true)
+        , timestamp(0)
     {
+        watcher = std::thread(&DataWatchdog::Run, this);
+        while (!facade)
```

Comment:
To have non-empty facade it must loaded in contructor and watcher thread must be started after the initial facade is loaded.